hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -269,23 +295,57 @@ def check_valid_minmax(prefix, val, info_obj):
             kw["exclude_max"] = exclude_max
 
         return st.floats(width=finfo.bits, **kw)
+    else:
+        # A less-inelegant solution to support complex dtypes exists, but as
+        # this is currently a draft feature, we might as well wait for
+        # discussion of complex inspection to resolve first - a better method
+        # might become available soon enough.
+        # See https://github.com/data-apis/array-api/issues/433
+        for attr in ["float32", "float64", "complex64"]:
+            if not hasattr(xp, attr):
+                raise NotImplementedError(
+                    f"Array module {xp.__name__} has no dtype {attr}, which is "
+                    "currently required for xps.from_dtype() to work with "
+                    "any complex dtype."
+                )
+        # Ideally we would infer allow_subnormal with a complex array here, just
+        # in case the array library has different FTZ behaviour between complex
+        # and float arrays. Unfortunately the spec currently has no mechanism to
+        # extract both real and imaj components, but should in the future.
+        # See https://github.com/data-apis/array-api/pull/427
```

Comment:
Also FYI there's no `x.__complex__()`, just wrote an issue for this https://github.com/data-apis/array-api/issues/486

Reply:
> in case the array library has different FTZ behaviour between complex and float arrays.

I don't think this is possible - the FTZ state is global, so unless the library changes that between every floating-point operation (depending on whether it's part of a complex array or not), you'll always have the same behaviour.